File: src/main/java/pirateboat/multifileHoster/MultifileHoster.java
package pirateboat.multifileHoster;

import pirateboat.torrent.Torrent;
import pirateboat.torrent.TorrentFile;

import java.util.ArrayList;
import java.util.List;

public interface MultifileHoster {

    String addTorrentToQueue(Torrent toBeAddedTorrent);
    List<Torrent> getRemoteTorrents();
    void enrichCacheStateOfTorrents(List<Torrent> torrents);
    void delete(Torrent remoteTorrent);
    List<TorrentFile> getFilesFromTorrent(Torrent torrent);
    int getPrio();
    String getName();
}

File: src/main/java/pirateboat/multifileHoster/MultifileHoster.kt
package pirateboat.multifileHoster

import pirateboat.torrent.Torrent
import pirateboat.torrent.TorrentFile

interface MultifileHoster {
    fun addTorrentToQueue(toBeAddedTorrent: Torrent): String
    fun getRemoteTorrents(): List<Torrent>
    fun enrichCacheStateOfTorrents(torrents: List<Torrent>)
    fun delete(remoteTorrent: Torrent)
    fun getFilesFromTorrent(torrent: Torrent): List<TorrentFile>
    fun getPrio(): Int
    fun getRemainingTrafficInMB(): Double
    fun getName(): String
}